src/server.rs

```diff
@@ -35,13 +38,16 @@ pub struct ErrorResponse {
 // NOTE: this should be used for sending the API response
 #[derive(Serialize, Debug, Clone)]
 pub struct DataResponse {
-    data: indexmap::IndexMap<String, serde_json::Value>,
+    data: IndexMap<String, serde_json::Value>,
 }
 
 pub async fn metadata_handler(payload: actix_web::web::Bytes) -> actix_web::HttpResponse {
-    let parse_result = json::parse(std::str::from_utf8(&payload).unwrap());
+    let payload_to_str = std::str::from_utf8(&payload);
+    let parse_result = match payload_to_str {
+        Ok(r) => json::parse(r),
+        Err(_err) => Err(json::Error::FailedUtf8Parsing),
+    };
 
-    // FIXME?: is this even necessary?
     let body: json::JsonValue = match parse_result {
         Ok(v) => v,
         Err(e) => json::object! { "error" => e.to_string() },
```

Comment:
Not part of this PR: but we can refactor this:
```suggestion
   let body = parse_result.unwrap_or_else(|e| json::object! { "error" => e.to_string() });
```

Reply:
Since there's some clean up being done in the PR itself, I think it should be fine. I can create a function for this purpose too, since we might be doing this in all other endpoints that consume JSON as request payloads.